src/services/index.ts

```diff
@@ -0,0 +1,448 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// @ts-ignore
+import hathorLib from '@hathor/wallet-lib';
+import { Event, Context, FullNodeEvent } from '../machines/types';
+import {
+  TxOutputWithIndex,
+  StringMap,
+  TokenBalanceMap,
+  TxInput,
+  Wallet,
+  DbTxOutput,
+  DbTransaction,
+  LastSyncedEvent,
+} from '../types';
+import {
+  prepareOutputs,
+  getAddressBalanceMap,
+  getUnixTimestamp,
+  unlockUtxos,
+  unlockTimelockedUtxos,
+  prepareInputs,
+  markLockedOutputs,
+  getTokenListFromInputsAndOutputs,
+  getWalletBalanceMap,
+  validateAddressBalances,
+} from '../utils';
+import {
+  getDbConnection,
+  addOrUpdateTx,
+  addUtxos,
+  updateTxOutputSpentBy,
+  updateAddressTablesWithTx,
+  getTransactionById,
+  getUtxosLockedAtHeight,
+  addMiner,
+  storeTokenInformation,
+  getLockedUtxoFromInputs,
+  incrementTokensTxCount,
+  getAddressWalletInfo,
+  generateAddresses,
+  addNewAddresses,
+  updateWalletTablesWithTx,
+  voidTransaction,
+  updateLastSyncedEvent as dbUpdateLastSyncedEvent,
+  getLastSyncedEvent,
+  getTxOutputsFromTx,
+  markUtxosAsVoided,
+  cleanupVoidedTx,
+} from '../db';
+import getConfig from '../config';
+import logger from '../logger';
+
+export const metadataDiff = async (_context: Context, event: Event) => {
+  const mysql = await getDbConnection();
+
+  try {
+    const fullNodeEvent = event.event as FullNodeEvent;
+    const {
+      hash,
+      metadata: { voided_by, first_block },
+    } = fullNodeEvent.event.data;
+    const dbTx: DbTransaction | null = await getTransactionById(mysql, hash);
+
+    if (!dbTx) {
+      if (voided_by.length > 0) {
+        // No need to add voided transactions
+        return {
+          type: 'IGNORE',
+          originalEvent: event,
+        };
+      }
+
+      return {
+        type: 'TX_NEW',
+        originalEvent: event,
+      };
+    }
+
+    // Tx is voided
+    if (voided_by.length > 0) {
+      // Was it voided on the database?
+      if (!dbTx.voided) {
+        return {
+          type: 'TX_VOIDED',
+          originalEvent: event,
+        };
+      }
+
+      return {
+        type: 'IGNORE',
+        originalEvent: event,
+      };
+    }
+
+    // Tx was voided in the database but is not anymore
+    if (dbTx.voided && voided_by.length <= 0) {
+      return {
+        type: 'TX_UNVOIDED',
+        originalEvent: event,
+      };
+    }
+
+    if (first_block
+       && first_block.length
+       && first_block.length > 0) {
+      if (!dbTx.height) {
+        return {
+          type: 'TX_FIRST_BLOCK',
+          originalEvent: event,
+        };
+      }
```

Comment:
Since we're removing all traces from it and then re-adding it from the data that came in the event, it should not be a problem, the transaction will be added with the height